### DISCUSSION TITLE
Panasonic Grid-EYE sensor (amg88xx)

### DISCUSSION THREAD

**Discussion by pideu-sj** [zephyrproject-rtos/zephyr#47690] [General]
We are currently developing an Arduino shield for the Panasonic Grid-EYE sensor, which is already included as the "amg88xx" sensor in Zephyr which is great.
There is also a sample project already available in `zephyr/samples/sensor/amg88xx` which refers to the "Adafruit AMG8833 8x8 Thermal Camera Sensor" board.

The documentation says: "This sample requires a board which provides a configuration for Arduino connectors and defines node aliases for the I2C and GPIO interfaces.
For more info about node structure and interrupt pin wiring see :zephyr_file:`samples/sensor/amg88xx/app.overlay`"

`app.overlay` looks like this:

```
&arduino_i2c {
	status = "okay";

	amg88xx@68 {
		compatible = "panasonic,amg88xx";
		reg = <0x68>;
		label = "AMG88XX";
		/* Pin D6 from Arduino Connector */
		int-gpios = <&arduino_header 12 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
		status = "okay";
	};
};
```

The complete I2C handling is hidden behind the arduino_i2c abstraction, which again is great.

But unfortunately the GPIO interrupt handling here is somewhat hardcoded.

The Adafruit sensor board does not use the Arduino header at all, but the documentation suggests that pin 12 is used for interrupt handling, so this now the de-facto standard.

Bad luck for us, because we chose pin 8 for this purpose, which is an equally good choice we guess.

We'd like to run that sample on our PAN1781 evaluation board which was recently added to the Zephyr list of boards.

In order to get the sample application for us to work we now have to modify the app.overlay directly like this:

`int-gpios = <&arduino_header 8 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;`

This is somehow ugly, because the PAN1781 supports the standard Arduino header and all it would have to know is where the new Panasonic Grid-EYE shield has the interrupt GPIO pin mapped to.

However, in theory this has nothing to do with the PAN1781 EVB we like to run the sample on, nor with the amg88xx sensor driver - this information belongs to the Panasonic Grid-EYE shield.

Now the question is: how does Zephyr handle this Arduino shield-specific interrupt GPIO configuration?

Is there an "Arduino shield layer" missing that sits between the amg88xx driver and the samples/sensor/amg88xx sample application that would be able to provide this kind of additional information or is there something we overlooked here.

Any suggestions how to handle this maybe new configuration problem is highly appreciated.

**Comment by jfischer-no** [comment]
app.overlay is for GridEYE Evaluation Kit (Ver 3.0). Kit should be configured to be a "shield", see README.rst, and it uses `Pin D6 from Arduino Connector`. I had plans to add shield support for it, and remove app.overlay. This could also have different overlays.

**Reply by pideu-sj** [reply]
@jfischer-no  Thanks for your fast reply!

> I had plans to add shield support for it, and remove app.overlay. This could also have different overlays.

That would be great. That means that we could simply create an own compatible for our own overlay, that we can add to the folder and a user can select, which one fits for his setup?

You said you "had" plans. Are they still existing? Let us know if we could help on this task somehow.

**Reply by jfischer-no** [reply]
Please loot at https://github.com/zephyrproject-rtos/zephyr/pull/47739

**Reply by pideu-sj** [reply]
That was fast! Almost a little bit to fast for us, because our shield will go in production in the next weeks. :-)

Thanks a lot you for your action, so far!

As soon as our shield is available, we will open a pull request and add our overlay to the boards/shields/amg88xx folder and extend the index.rst documentation to our shield.
